packages/jira-adapter/test/client/connection.test.ts

```diff
@@ -28,34 +28,67 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
-import axios from 'axios'
+import axios, { AxiosRequestConfig } from 'axios'
 import MockAdapter from 'axios-mock-adapter'
+import { client as clientUtils } from '@salto-io/adapter-components'
+import _ from 'lodash'
 import { validateCredentials, createConnection } from '../../src/client/connection'
 
-describe('validateCredentials', () => {
-  let mockAxios: MockAdapter
-  let result: string
-  beforeEach(async () => {
-    mockAxios = new MockAdapter(axios)
-    mockAxios.onGet().reply(200, { baseUrl: 'http://my.jira.net' })
-    const connection = await createConnection({ retries: 1 }).login(
-      { baseUrl: 'http://myJira.net', user: 'me', token: 'tok' }
-    )
-    result = await validateCredentials({ connection })
-  })
-  afterEach(() => {
-    mockAxios.restore()
-  })
 
-  it('should get server info with auth headers', () => {
-    expect(mockAxios.history.get).toHaveLength(1)
-    const request = mockAxios.history.get[0]
-    expect(request.url).toEqual('/rest/api/3/serverInfo')
-    expect(request.baseURL).toEqual('http://myJira.net')
-    expect(request.auth).toEqual({ username: 'me', password: 'tok' })
-  })
+describe('Jira connection', () => {
+  describe('validateCredentials', () => {
+    let mockAxios: MockAdapter
+    let connection: clientUtils.APIConnection
+
+    beforeEach(async () => {
+      mockAxios = new MockAdapter(axios)
+      mockAxios.onGet('/rest/api/3/configuration').reply(200)
+      mockAxios.onGet('/rest/api/3/serverInfo').reply(200, { baseUrl: 'http://my.jira.net' })
+      connection = await createConnection({ retries: 1 }).login(
+        { baseUrl: 'http://myJira.net', user: 'me', token: 'tok' }
+      )
+    })
+    afterEach(() => {
+      mockAxios.restore()
+    })
+
+    describe('pass', () => {
+      let result: string
+
+      const getServerInfoRequest = (): AxiosRequestConfig => {
+        const request = mockAxios.history.get.find(r => r.url === '/rest/api/3/serverInfo')
+        if (_.isUndefined(request)) {
```

Comment:
should not use `_` when there is a built in alternative that is equivalent (it is just a bad habit, `_` has a lot of redundant code that deals with supporting different runtimes that we just do not need)
```suggestion
        if (request === undefined) {
```

Reply:
V,

Also, what do you think about the following scenario though? (Just curious since I've done something similar in one of my previous PRs)

```typescript
const hasNullValue = (array: unknown[]): boolean => array.some(v => === null)
// Alternative using lodash
const hasNullValue = (array: unknown[]): boolean => array.some(_.isNull)
```

Reply:
lodash should be used when it makes the code significantly more readable
the "significantly" part is vague and is left up to personal opinion, personally, I think `.some(v => v === null)` is slightly worse but not significantly so I'd prefer it, but not critical